File: client/src/types/transaction.ts
interface Transaction {
  id: string;
  amount: number;
  date: Date;
  category: string;
  subcategory: string;
  merchantName: string;
  pending: boolean;
  deleted: Date;
  source: string;
  accountID: string;
}

interface NewTransaction extends Partial<Transaction> {}

export type { Transaction, NewTransaction };

File: server/BudgetBoard-Database/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace BudgetBoard.Database.Models;

public class Transaction
{
    public Guid ID { get; set; }
    public string? SyncID { get; set; }
    public decimal Amount { get; set; }
    public DateTime Date { get; set; }
    [DisplayFormat(NullDisplayText = "No Category")]
    public string? Category { get; set; }
    [DisplayFormat(NullDisplayText = "No Subcategory")]
    public string? Subcategory { get; set; }
    [DisplayFormat(NullDisplayText = "No Merchant")]
    public string? MerchantName { get; set; }
    public bool Pending { get; set; } = false;
    public DateTime? Deleted { get; set; } = null;
    public required string Source { get; set; }
    public required Guid AccountID { get; set; }
    public Account? Account { get; set; } = null!;
}
